packages/legend-application/src/components/shared/BasicValueSpecificationEditor.tsx

```diff
@@ -170,23 +171,83 @@ const StringPrimitiveInstanceValueEditor = observer(
     className?: string | undefined;
     setValueSpecification: (val: ValueSpecification) => void;
     resetValue: () => void;
+    valueOptions?:
```

Comment:
maybe `selectorConfig: { values: string[] | undefined; isLoading: boolean; reloadValues: (val: string) => void }`. From API perspective, it makes more sense to have the updater function takes the input value.